apps/platform/trpc/routers/authRouter/passwordRouter.ts

```diff
@@ -29,16 +29,84 @@ export const passwordRouter = router({
       const { db, user } = ctx;
       const userId = user.id;
 
+      const userData = await db.query.users.findFirst({
+        where: eq(users.id, userId),
+        columns: {
+          id: true,
+          publicId: true,
+          username: true,
+          createdAt: true
+        },
+        with: {
+          account: {
+            columns: {
+              passwordHash: true
+            },
+            with: {
+              authenticators: {
+                columns: {
+                  nickname: true
+                }
+              }
+            }
+          }
+        }
+      });
+
+      if (!userData) {
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: 'User not found'
+        });
+      }
+
+      // Check if password or passkey is not set and timeTillOrphaned is passed
+      if (
+        !(
+          userData.account.passwordHash ||
+          userData.account.authenticators.length > 0
+        ) &&
+        new Date().getTime() >=
+          userData.createdAt.getTime() + useRuntimeConfig().timeTillOrphanedUser
+      ) {
+        // remove orphaned user before returning
+        await db.delete(users).where(eq(users.id, userData.id));
+
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: 'User not found'
+        });
+      }
```

Comment:
do we want to do this here?
if a user signs up, but does not create a password, this will never be hit
it will only be hit if the user returns more than 30 minutes later and tries to set a password

Reply:
we can extract out the repetitive logic, but that code it there to prevent user from creating a password 30 minutes later if they haven't created one already. We could make the message more verbose like `Your account has been deleted as you didn't set your password/passkey in required time period, please sign up again`

Reply:
Do you have a better plan for this?
I made the message more verbose, suggesting what happened to the user.
If we remove orphan accounts from time to time then users will not have to see it anyways.

Reply:
yeah more verbose message is good enough for now